dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/query/operators/EmptyOperator.java

```diff
@@ -0,0 +1,68 @@
+package org.hisp.dhis.query.operators;
+
+/*
+ *
+ *  Copyright (c) 2004-2017, University of Oslo
+ *  All rights reserved.
+ *
+ *  Redistribution and use in source and binary forms, with or without
+ *  modification, are permitted provided that the following conditions are met:
+ *  Redistributions of source code must retain the above copyright notice, this
+ *  list of conditions and the following disclaimer.
+ *
+ *  Redistributions in binary form must reproduce the above copyright notice,
+ *  this list of conditions and the following disclaimer in the documentation
+ *  and/or other materials provided with the distribution.
+ *  Neither the name of the HISP project nor the names of its contributors may
+ *  be used to endorse or promote products derived from this software without
+ *  specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ *  ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ *  WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ *  DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ *  ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ *  (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ *  ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ *  (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ *  SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ */
+
+import org.hibernate.criterion.Criterion;
+import org.hibernate.criterion.Restrictions;
+import org.hisp.dhis.query.Typed;
+import org.hisp.dhis.query.planner.QueryPath;
+
+import java.util.Collection;
+
+/**
+ * @author Viet Nguyen <viet@dhis2.org>
+ */
+public class EmptyOperator extends Operator
+{
+    public EmptyOperator()
+    {
+        super( "empty", Typed.from( Collection.class ) );
+    }
+
+    @Override
+    public Criterion getHibernateCriterion( QueryPath queryPath )
+    {
+
+        return Restrictions.sizeEq( queryPath.getPath(),0 );
+    }
+
+    @Override
+    public boolean test( Object value )
+    {
+        if ( value == null )
+        {
+            return false;
+        }
+        Collection<?> collection = (Collection<?>) value;
```

Comment:
how does this work if your are using the operator on a string? integer? have a look at the other operators to see how i validate them